azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,345 @@
+            }
+         }
+      }
+   }
+
+   @Override
+   public void rebootNode(final String id) {
+      api.getVirtualMachineApi(getGroupId()).restart(id);
+   }
+
+   @Override
+   public void resumeNode(final String id) {
+      api.getVirtualMachineApi(getGroupId()).start(id);
+   }
+
+   @Override
+   public void suspendNode(final String id) {
+      api.getVirtualMachineApi(getGroupId()).stop(id);
+   }
+
+   private List<PublicIPAddress> getIPAddresses(Deployment deployment) {
+      List<PublicIPAddress> list = new ArrayList<PublicIPAddress>();
+      String resourceGroup = getResourceGroupFromId(deployment.id());
+
+      if (deployment.properties() != null && deployment.properties().dependencies() != null) {
+         List<Deployment.Dependency> dependencies = deployment.properties().dependencies();
+         for (int d = 0; d < dependencies.size(); d++) {
+            if (dependencies.get(d).resourceType().equals("Microsoft.Network/networkInterfaces")) {
+               List<Deployment.Dependency> dependsOn = dependencies.get(d).dependsOn();
+               for (int e = 0; e < dependsOn.size(); e++) {
+                  if (dependsOn.get(e).resourceType().equals("Microsoft.Network/publicIPAddresses")) {
+                     String resourceName = dependsOn.get(e).resourceName();
+                     PublicIPAddress ip = api.getPublicIPAddressApi(resourceGroup).get(resourceName);
+                     list.add(ip);
+                  }
+               }
+            }
+         }
+      }
+      return list;
+   }
+
+   @Override
+   public Iterable<VMDeployment> listNodes() {
+      System.out.println("listNodes");
+
+      List<Deployment> deployments = api.getDeploymentApi(getGroupId()).list();
+
+      System.out.println("Found " + deployments.size() + " nodes");
+      List<VMDeployment> vmDeployments = new ArrayList<VMDeployment>();
+
+      for (Deployment d : deployments){
+         VMDeployment vmDeployment = new VMDeployment();
+         vmDeployment.deployment = d;
+         vmDeployment.vm = api.getVirtualMachineApi(getGroupId()).getInstanceDetails(d.name());
+         List<PublicIPAddress> list = getIPAddresses(d);
+         vmDeployment.ipAddressList = list;
+         vmDeployments.add(vmDeployment);
+      }
+      return vmDeployments;
+   }
+
+   @Override
+   public Iterable<VMDeployment> listNodesByIds(final Iterable<String> ids) {
+      System.out.println("listNodesByIds");
+      for (String str : ids) {
+         System.out.println(str);
+      }
+      return Iterables.filter(listNodes(), new Predicate<VMDeployment>() {
+         @Override
+         public boolean apply(final VMDeployment input) {
+            return Iterables.contains(ids, input.deployment.name());
+         }
+      });
+   }
+
+   private String getGroupId() {
+      String group =  System.getProperty("test.azurecompute-arm.groupname");
+      if (group == null)
+         group = "jCloudsGroup";
+      return group;
+   }
```

Comment:
Hi, My old comment from our slack below. IMHO it is good to have it jCloudsGroup as default and let users to config that if they want. If we use jClouds groups as resource groups, some of the jClouds-features are not working as we can work in one resource group per time.

Example: if we create 10 VMs and we name them "machine", there will be jClouds group called "machine" and VMs will be named like machine-e3e, machine-e3f, machine-e3g etc

[9:28] 
if we then create another 10 VMs and name them machine2, there will be new group machine2 etc

[9:29] 
if we use this machine2 as azure resource group name, the jClouds API will be working incorrectly

[9:29] 
Because, there is function that we can use to run script in all machines matching some regex

[9:30] 
if we say "runScriptInAllMachinesMatching("machine")", we should run that script in machine-group and machine2-group.

[9:31] 
and basically jClouds will then call our getAllNodes once

[9:31] 
and we have implemented getAllNodes so that it will return nodes only from one resourceGroup